standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5934,6 +5936,12 @@ private void alter_table_core(String catName, String dbname, String name, Table
       catName = MetaStoreUtils.getDefaultCatalog(conf);
     }
 
+    // HIVE-25282: Drop/Alter table in REMOTE db should fail
+    Database db = get_database_core(catName, dbname);
```

Comment:
per the comment above, can you wrap this code in a try/catch and re-throw the NoSuchObjectException as an InvalidOperationException? Ideally this should throw back a NoSuchObjectException, but I am concerned about backward compatibilty with older clients.